taskvine/src/manager/vine_schedule.c

```diff
@@ -86,6 +67,57 @@ int vine_schedule_in_ramp_down(struct vine_manager *q)
 	return 0;
 }
 
+/* Check if worker resources are enough to run the task.
+ * @param q     Manager info structure.
+ * @param w     Worker info structure.
+ * @param t     Task info structure.
+ * @param tr    Task resources.
+ * @return 1 if yes, 0 otherwise. */
+int check_worker_have_enough_resources(
+		struct vine_manager *q, struct vine_worker_info *w, struct vine_task *t, struct rmsummary *tr)
+{
+	struct vine_resources *worker_net_resources = vine_resources_copy(w->resources);
+
+	/* Pretend to reclaim resources from empty libraries. */
```

Comment:
This is potentially an expensive loop, so make it conditional on `t->needs_library`.
And then break it out into a separate function something like `subtract_empty_libraries` or similar.

Reply:
I think this check is needed for any type of task, as a regular task can kick out library tasks if they are empty.

Reply:
Ah, I see you are right.  So we should be cognizant that this change increases the complexity of the inner scheduling loop.  And may have an impact on scheduling large numbers of tasks.  @colinthomas-z80 what do you think?

Reply:
Do we have a script that benchmarks TaskVine against a large number of tasks? That'd be helpful here.

Reply:
I usually use vine_benchmark with something like `submit 0 0 0 100000`

Reply:
Discuss with the person sitting next to you!

Reply:
And if I understand this correctly, this iterates through currently running tasks rather than queued? 

I think it is safe to say that most of the time there are not a hugely significant amount of tasks running.

Reply:
@dthain Office hours :)